src/chromium/crNetworkManager.ts

```diff
@@ -165,7 +166,16 @@ export class CRNetworkManager {
         redirectedFrom = request.request;
       }
     }
-    const frame = event.frameId ? this._page._frameManager.frame(event.frameId) : workerFrame;
+    let frame = event.frameId ? this._page._frameManager.frame(event.frameId) : workerFrame;
+
+    // Check if it's main resource request interception (targetId === main frame id).
+    if (!frame && interceptionId && event.frameId === CRTarget.fromPage(this._page)._targetId) {
+      // Main resource request for the page is being intercepted so the Frame is not created
+      // yet. Precreate it here for the purposes of request interception. It will be updated
+      // later as soon as the request contnues and we receive frame tree from the page.
+      frame = this._page._frameManager.frameAttached(event.frameId, null);
```

Comment:
I am a bit worried about this (user can call `request.frame().url()` and be surprised, or even try to evaluate in it), but oh well.

Reply:
Let's see, if it becomes a problem we can put additional protection to ensure the page will throw meaningful errors.